integration-libs/cdc/root/guards/gigya-raas.guard.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { Router, UrlTree } from '@angular/router';
+import { GigyaRaasComponentData } from '@spartacus/cdc/core';
+import {
+  AuthRedirectService,
+  AuthService,
+  CmsService,
+  PageContext,
+  RoutingService,
+  SemanticPathService,
+  isNotUndefined,
+} from '@spartacus/core';
+import { Observable, catchError, filter, map, of, switchMap, take } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GigyaRaasGuard {
+  protected routingService = inject(RoutingService);
+  protected cmsService = inject(CmsService);
+  protected authService = inject(AuthService);
+  protected authRedirectService = inject(AuthRedirectService);
+  protected router = inject(Router);
+  protected semanticPathService = inject(SemanticPathService);
+
+  canActivate(): Observable<boolean | UrlTree> {
+    return this.routingService.getNextPageContext().pipe(
+      filter(isNotUndefined),
+      take(1),
+      switchMap((pageContext) =>
+        this.getComponentsByType(pageContext, 'GigyaRaasComponent')
+      ),
+      switchMap((components) => this.getComponentData(components)),
```

Comment:
[NITPICK]
For brevity of the `canActivate()`, what about extracting this logic to a separate method `getComponentData()` ?

Reply:
done